gqlalchemy/models.py

```diff
@@ -57,7 +61,36 @@ def to_cypher(self) -> str:
         return f"(n:{self.label}) ASSERT EXISTS (n.{self.property})"
 
 
-class GraphObject:
+class GraphObject(BaseModel):
+    _subtypes_ = dict()
+
+    def __init_subclass__(cls, type=None):
+        cls._subtypes_[type or cls.__name__.lower()] = cls
+
+    @classmethod
+    def __get_validators__(cls):
+        yield cls._convert_to_real_type_
+
+    @classmethod
+    def _convert_to_real_type_(cls, data):
+        data_type = data.get("type")
+
+        sub = cls if data_type is None else cls._subtypes_.get(data_type)
+
+        if sub is None:
+            warnings.warn(  # GQLAlchemy failed to find a subclass. #  )
```

Comment:
I added this comment because the default warning formatting prints the line that threw the warning. Before it was just:
```
warnings.warn(
```

Long term solution is to create a custom formatting function.